pkg/validation/spec/header.go

```diff
@@ -53,7 +53,11 @@ func (h Header) MarshalJSON() ([]byte, error) {
 	if err != nil {
 		return nil, err
 	}
-	return swag.ConcatJSON(b1, b2, b3), nil
+	b4, err := json.Marshal(h.VendorExtensible)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
are we using json marshaling of this type anywhere in kubernetes today (in CRD validation, or openapi v2 generation, etc)? just wanting to make sure this isn't an API-facing change in current use

Reply:
Looking at the instances of `Response` through https://raw.githubusercontent.com/kubernetes/kubernetes/master/api/openapi-spec/swagger.json none of the default types define headers for their responses.

In the case of CRDs, it may be possible that CRDs populate response headers for their open api spec. I am wholly unfamiliar with that part of the infrastructure, though. @apelisse do you know whether existing CRDs are able to populate `Schema.Paths.Responses.Headers`?

Reply:
CRDs don't have the ability to do anything specific to their response header, if I understand correctly. I don't think it's being used at all.

Also, I'm not sure what this would break, or how this would be an api-facing change.

Reply:
yeah, I swept k8s.io/kubernetes and couldn't find any places we construct this

> Also, I'm not sure what this would break, or how this would be an api-facing change.

just wanted to know if this was going to start including directives places we hadn't before, and think if any known clients would be unable to handle those (think `kubectl`'s fragile handling of openapi v2)